### PR TITLE
protocol/validation: add ValidateBlock unit tests

### PR DESCRIPTION
**Pull request by kr**
These are not comprehensive, but it's a start.

### CODE REVIEW

**Comment by kr** [comment]
PTAL @bobg

File: protocol/block.go [hunk 1]
```diff
@@ -48,6 +48,10 @@ func (c *Chain) GetBlock(ctx context.Context, height uint64) (*bc.Block, error)
 // After generating the block, the pending transaction pool will be
 // empty.
 func (c *Chain) GenerateBlock(ctx context.Context, prev *bc.Block, snapshot *state.Snapshot, now time.Time, txs []*bc.Tx) (*bc.Block, *state.Snapshot, error) {
+	// TODO(kr): move this into a lower-level package (e.g. chain/protocol/bc)
```
Comment (bobg) [inline protocol/block.go]: :+1:

**Comment by bobg** [comment]
LGTM

**Comment by kr** [comment]
(note I deliberately canceled the wercker run so that I can test #839)

**Pull request merged**